src/library/scala/collection/immutable/IndexedSeqView.scala

```diff
@@ -0,0 +1,416 @@
+    *   underlying:
+    *     The main collection this is a slice of.
+    *     Offsets within the underlying collection are also maintained, these are `lo` and `hi`
+    *   prefix: The IndexedSeqView of elements which precedes `underlying`, usually obtained through prepending
+    *   suffix: The IndexedSeqView of elements which follows `underlying`, usually obtained through appending
+    *
+    * The overall picture is then:
+    *
+    * prefix                           underlying                          suffix
+    * [*******************************][     *********************        ][*******************************]
+    *                                  |----|lo
+    *                                  |--------------------------|hi
+    *
+    *
+    * Also tracked are:
+    *   isReversed:
+    *     a flag that tracks whether at read time, the elements should be read:
+    *       right to left (true)
+    *       left to right (false)
+    *
+    *     The flag will not be set, unless a previous `reverse` transformation has been applied
+    *
+    *   mapFunctions:
+    *     a queue of functions, possibly of different types, that are applied to each element
+    *     from left to right at read time. This is used as a stack-safe alternative to arbitrily
+    *     nested functions during repeated calls to .map(f) transformations.
+    *
+    * @param underlying The main sequence of elements over which this slice manages a view of. Will not typically be
+    *                   empty, but there is no breakage caused if it is empty
+    * @param prefix the elements which precede the elements in underlying
+    * @param from the offset from the left, of elements in underlying. This is the lowest index which is included
+    * @param until the offset from the left, of elements in underlying. This is the lowest index which is EXCLUDED
+    * @param suffix the elements which succeed the elements in underlying
+    * @param isReversed if true, then this slice is to be read right-to-left. Otherwise, left-to-right
+    * @param mapFunctions an IndexedSeq of functions which represent previous mapping transformations, to be applied
+    *                     left-to-right, when this view is strictly evaluated.
+    *
+    *                     Care is taken to ensure that the first input type of the first map function
+    *                     is of type `A`, and the last has output type `B`
+    *
+    * @tparam A the type of element in underlying
+    * @tparam B the type of element this is a view of, which are obtained by mapping elements of type `A` through
+    *           mapFunctions
+    */
+  @SerialVersionUID(3L)
+  private[scala] final class Slice[+A, +B](
+    prefix: IndexedSeqView[B],
+    underlying: SomeIndexedSeqOps[A],
+    from: Int,
+    until: Int,
+    suffix: IndexedSeqView[B],
+    isReversed: Boolean,
+    mapFunctions: IndexedSeq[Any => Any]) extends AbstractIndexedSeqView[B] {
```

Comment:
I take it the runtime type here would typically be `Vector`. That gives us decent performance in in `Slice.map` that appends, and good performance in `Slice.apply`.

Reply:
That's correct, I'll change this to specify Vector